### PR TITLE
use _doGetTSOEndpoints() to set all tso endpoints

### PR DESCRIPTION
**Pull request by dlifw**
Use _doGetTSOEndpoints() in the TSO client to set all the TSO endpoints.

### CODE REVIEW

File: src/k2/tso/client/Client.cpp [hunk 1]
```diff
@@ -71,7 +71,20 @@ seastar::future<> TSOClient::_doGetTSOEndpoints(dto::GetTSOEndpointsRequest &req
         }
         _tsoErrorBound = resp.minTransTime;
         _tsoServerEndpoint = RPC().getTXEndpoint(resp.endpoints[0]);
```
Comment (jfunston) [inline src/k2/tso/client/Client.cpp:73]: _tsoServerEndpoint isn't necessary anymore is it?

File: src/k2/tso/service/Service.cpp [hunk 2]
```diff
@@ -158,12 +158,6 @@ seastar::future<bool> TSOService::_assign(uint64_t tsoID, k2::Duration errBound)
             K2LOG_I(log::tsoserver, "Error bound validation succeeded? {}", isValidErrorBound);
```
Comment (iafuture) [inline src/k2/tso/service/Service.cpp:158]: Also, remove the _collectWorkerURLs method and usage just above, as well as the _workersURLs member var

**Review by iafuture — CHANGES_REQUESTED** [review]

File: src/k2/tso/client/Client.cpp (hunk 1 above)
Comment (dlifw) [reply src/k2/tso/client/Client.cpp]: Removed.

File: src/k2/tso/service/Service.cpp (hunk 2 above)
Comment (dlifw) [reply src/k2/tso/service/Service.cpp:146]: Removed.

**Review by iafuture — APPROVED** [review]

**Review by jfunston — APPROVED** [review]

**Pull request merged**